# Water Potability: Support Vector Machine Classification

## Project Overview

This notebook applies a Support Vector Classifier to predict whether a water sample is potable from nine measured water-quality attributes.

The model uses the default `SVC()` configuration and a fixed 70/30 train/test split so the supplied result can be reproduced directly.

### Technical Coverage

- tabular binary classification
- Support Vector Classifier
- train/test splitting
- accuracy
- confusion-matrix analysis
- class-balance interpretation

## 1. Environment

In [1]:
import pandas as pd
import numpy as np
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn.model_selection import train_test_split


## 2. Water Potability Dataset

The dataset contains **2,011 complete observations**, nine predictor variables, and a binary `Potability` target.

| Feature | Description |
|---|---|
| `ph` | acidity/alkalinity |
| `Hardness` | water hardness |
| `Solids` | total dissolved solids |
| `Chloramines` | chloramine concentration |
| `Sulfate` | sulfate concentration |
| `Conductivity` | electrical conductivity |
| `Organic_carbon` | organic carbon |
| `Trihalomethanes` | trihalomethane concentration |
| `Turbidity` | turbidity |
| `Potability` | binary target |

### Data Loading and Inspection

In [4]:
from google.colab import files
import io

uploaded = files.upload()

filename = next(iter(uploaded))

water_df = pd.read_excel(
    io.BytesIO(uploaded[filename])
)

water_np = water_df.to_numpy()

X = water_np[:, :-1]

y = water_np[:, -1]

print("Dataset shape:", water_np.shape)

print("Feature matrix shape:", X.shape)

print(
    "Class counts [not potable, potable]:",
    np.bincount(y.astype(int))
)

water_df.head()


Saving water_potability.xlsx to water_potability.xlsx
Dataset shape: (2011, 10)
Feature matrix shape: (2011, 9)
Class counts [not potable, potable]: [1200  811]


,ph,Hardness,Solids,Chloramines,Sulfate,Conductivity,Organic_carbon,Trihalomethanes,Turbidity,Potability
0,7.899452,210.734124,15896.365940,6.907203,319.886957,448.666423,18.169921,124.000000,2.853767,1
1,6.145148,197.541072,39657.272110,9.900159,288.157883,319.434033,11.587378,120.030077,4.600886,0
2,5.036454,190.164520,29258.738140,4.991061,300.475925,332.359715,11.055801,116.161622,3.534665,1
3,8.969697,195.744765,9049.682595,7.467068,396.453568,378.528511,17.757697,114.208671,3.983099,0
4,8.285072,151.573778,14402.726700,9.050080,303.081838,322.521815,13.652653,114.034946,4.274661,1


## 3. Train/Test Split

The first nine columns are used as predictors and `Potability` is used as the target. A 30% test split with `random_state=4361` reproduces the supplied evaluation.

In [5]:
x_train, x_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.3,
    random_state=4361
)

print("Training shape:", x_train.shape)
print("Test shape:", x_test.shape)


Training shape: (1407, 9)
Test shape: (604, 9)


## 4. Default Support Vector Classifier

The baseline uses the default `SVC()` configuration.

In [6]:
model = SVC()
model.fit(x_train, y_train)
pred = model.predict(x_test)
accuracy = accuracy_score(y_test, pred)
print(f'Accuracy {accuracy:.4}')
print(confusion_matrix(y_test, pred))

Accuracy 0.601
[[363   0]
 [241   0]]


## 5. Result Interpretation

The test split contains 363 non-potable samples and 241 potable samples. The confusion matrix is:

```text
[[363   0]
 [241   0]]
```

The fitted baseline predicts every test observation as class 0 (not potable). Its accuracy of **0.6010** is therefore driven by the majority class rather than successful separation of the two classes.

This result provides a clear baseline for later SVM experiments involving feature scaling, kernel selection, or parameter tuning.

## 6. Key Findings

- The default SVC reaches **0.6010** test accuracy on the supplied split.
- The confusion matrix shows that the model predicts only the majority class.
- Accuracy alone would hide this behavior; the confusion matrix is necessary for interpreting the classifier.
- The experiment demonstrates why preprocessing and model configuration can be especially important for SVMs on features measured at very different numerical scales.